asreview/webapp/auth.py

```diff
@@ -0,0 +1,144 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import datetime
+from pathlib import Path
+
+from flask import Blueprint
+from flask import jsonify
+from flask import request
+from flask_cors import CORS
+from flask_login import current_user, login_user, logout_user
+from sqlalchemy.exc import SQLAlchemyError
+
+from asreview.utils import asreview_path
+from asreview.webapp import DB
+from asreview.webapp.authentication.login_required import asreview_login_required
+from asreview.webapp.authentication.models import User
+
+bp = Blueprint('auth', __name__, url_prefix='/auth')
+CORS(
+    bp,
+    resources={r"*": {"origins": "http://localhost:3000"}},
+    supports_credentials=True,
+)
+
+
+@bp.route('/signin', methods=["POST"])
+def signin():
+    username = request.form.get('username').strip()
+    password = request.form.get('password')
+
+    # check if username already exists
+    user = User.query.filter(User.username == username).one_or_none()
+    # if the user exist proceed and verify
+    if not user:
+        result = (404, {'message': f'User account {username} does not exist.'})
+    else:
+        # verify password
+        if user.verify_password(password):
+            logged_in = login_user(
+                user,
+                remember=True,
+                duration=datetime.timedelta(days=31)
+            )
+            result = (200, {
+                'logged_in': logged_in,
+                'username': username,
+                'id': user.id
+            })
+        else:
+            # password is wrong
+            result = (
+                404,
+                {'message': f'Incorrect password for user {username}'}
+            )
+
+    status, message = result
+    # check in database if username exists and if password is correct
+    response = jsonify(message)
+    return response, status
+
+
+@bp.route('/signup', methods=["POST"])
+def signup():
+    print('-------->')
+    print(request.form)
+    username = request.form.get('username').strip().lower()
+    password = request.form.get('password')
+    first_name = request.form.get('first_name', '').strip()
+    last_name = request.form.get('last_name', '').strip()
+    affiliation = request.form.get('affiliation', '').strip()
+    email = request.form.get('email', '').strip()
+    public = bool(int(request.form.get('public', '0')))
+
+    # check if username already exists
+    user = User.query.filter(User.username == username).one_or_none()
```

Comment:
This will also expose the public flag of the user. Maybe I misunderstood the idea behind the public flag. It's more something like: Open for collaboration.